column_buffer_go18.go

```diff
@@ -178,6 +181,35 @@ func writeRowsFuncOfOptional(t reflect.Type, schema *Schema, path columnPath, wr
 	}
 }
 
+func writeRowsFuncOfTime(_ reflect.Type, schema *Schema, path columnPath) writeRowsFunc {
+	t := reflect.TypeOf(int64(0))
+	elemSize := uintptr(t.Size())
+	writeRows := writeRowsFuncOf(t, schema, path)
+
+	return func(columns []ColumnBuffer, rows sparse.Array, levels columnLevels) error {
+		if rows.Len() == 0 {
+			return writeRows(columns, rows, levels)
+		}
+
+		for i := 0; i < rows.Len(); i++ {
+			p := (*time.Time)(rows.Index(i))
```

Comment:
Would it be worth adding a specialization of `sparse.TimeArray` like we do for other types to improve type safety?

Reply:
Good catch. Would that code then look something like this?

```go
ts := rows.TimeArray()
for i := 0; i < ts.Len(); i++ {
	t := ts.Index(i)
	val := reflect.ValueOf(t.UnixNano())
	a := makeArray(unsafecast.PointerOfValue(val), 1, elemSize)
	...
}
```